service/src/main/java/bio/terra/catalog/rawls/RawlsService.java

```diff
@@ -14,21 +22,79 @@
 @Component
 public class RawlsService {
   private static final Logger logger = LoggerFactory.getLogger(RawlsService.class);
+  public static final List<String> ACCESS_LEVEL_LIST = List.of("accessLevel");
+  public static final List<String> ACCESS_LEVEL_AND_ID_LIST =
+      List.of("accessLevel", "workspace.workspaceId");
 
   private final RawlsConfiguration rawlsConfig;
   private final Client commonHttpClient;
+  private static final List<WorkspaceAccessLevel> OWNER_ROLES =
+      List.of(WorkspaceAccessLevel.PROJECT_OWNER, WorkspaceAccessLevel.OWNER);
+  private static final List<WorkspaceAccessLevel> WRITER_ROLES =
+      List.of(
+          WorkspaceAccessLevel.PROJECT_OWNER,
+          WorkspaceAccessLevel.OWNER,
+          WorkspaceAccessLevel.WRITER);
+  private static final List<WorkspaceAccessLevel> READER_ROLES =
+      List.of(
+          WorkspaceAccessLevel.PROJECT_OWNER,
+          WorkspaceAccessLevel.OWNER,
+          WorkspaceAccessLevel.WRITER,
+          WorkspaceAccessLevel.READER);
 
   @Autowired
   public RawlsService(RawlsConfiguration rawlsConfig) {
     this.rawlsConfig = rawlsConfig;
     this.commonHttpClient = new ApiClient().getHttpClient();
   }
 
+  private ApiClient getApiClient(AuthenticatedUserRequest user) {
+    ApiClient apiClient = getApiClient();
+    apiClient.setAccessToken(user.getToken());
+    return apiClient;
+  }
+
   private ApiClient getApiClient() {
     // Share one api client across requests.
     return new ApiClient().setHttpClient(commonHttpClient).setBasePath(rawlsConfig.basePath());
   }
 
+  public Map<String, List<String>> getWorkspaceIdsAndRoles(AuthenticatedUserRequest user) {
+    try {
+      return workspacesApi(user).listWorkspaces(ACCESS_LEVEL_AND_ID_LIST).stream()
+          .collect(
+              Collectors.toMap(
+                  workspaceListResponse -> workspaceListResponse.getWorkspace().getWorkspaceId(),
+                  workspaceListResponse ->
+                      List.of(workspaceListResponse.getAccessLevel().toString())));
```

Comment:
They are not the same as the TDR role names. I could see determining a unified set of roles. It looks like right now we have `Owner` and `Reader` from TDR, and `Owner`, `Writer`, and `Reader` from workspaces. My inclination is that this seems like the right breakdown to me, so my inclination is to do that here, and maybe make a ticket if we think those are the wrong names/to maybe consider user testing the role names to see what users think makes sense?

Reply:
Actually, what roles do we get back from the TDR? Are they these?
https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L1120

If so, I have no idea how those map to catalog roles offhand, and we may want to make a design/get some product input and user testing, because if we are going to show these on the UI, we would want to make sure that it makes sense to users.

Reply:
To be done in a follow up.